pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -55,44 +55,89 @@ let
+      [ "--add-flags"
+        # --cmd is needed even if we disable this ${prog} provider or not
+        ("\"--cmd " +
+          (if withProg then
+            # All host programs end up in the same path
+            "'let g:${prog}_host_prog=$out/bin/nvim-${prog}'"
+          else
+            # Tells nvim to practically not load this provider
+            "'let g:loaded_${prog}_provider=1'"
+          )
+        + "\"")
+      ]
+    ) hostprog_check_table))
+    ++ lib.optionals withRuby [
+      "--set" "GEM_HOME" "${rubyEnv}/${rubyEnv.ruby.gemPath}"
+    ]
+    ++ lib.optionals (binPath != "") [
+      "--suffix" "PATH" binPath
+    ]
+    # this relies on a patched neovim, see
+    # https://github.com/neovim/neovim/issues/9413
+    ++ lib.optionals (configure != {}) [
+      "--set" "NVIM_SYSTEM_RPLUGIN_MANIFEST" "$out/rplugin.vim"
+      "--add-flags" "-u" "${vimUtils.vimrcFile configure}"
+    ]
+  ;
+
   in
-  stdenv.mkDerivation {
+  symlinkJoin {
       name = "neovim-${stdenv.lib.getVersion neovim}";
-      buildCommand = let bin="${neovim}/bin/nvim"; in ''
-        if [ ! -x "${bin}" ]
-        then
-            echo "cannot find executable file \`${bin}'"
-            exit 1
-        fi
-
-        makeWrapper "$(readlink -v --canonicalize-existing "${bin}")" \
-          "$out/bin/nvim" --add-flags " \
-        --cmd \"${if withNodeJs then "let g:node_host_prog='${nodePackages.neovim}/bin/neovim-node-host'" else "let g:loaded_node_provider=1"}\" \
-        --cmd \"${if withPython then "let g:python_host_prog='$out/bin/nvim-python'" else "let g:loaded_python_provider = 1"}\" \
-        --cmd \"${if withPython3 then "let g:python3_host_prog='$out/bin/nvim-python3'" else "let g:loaded_python3_provider = 1"}\" \
-        --cmd \"${if withRuby then "let g:ruby_host_prog='$out/bin/nvim-ruby'" else "let g:loaded_ruby_provider=1"}\" " \
-        --suffix PATH : ${binPath} \
-        ${optionalString withRuby '' --set GEM_HOME ${rubyEnv}/${rubyEnv.ruby.gemPath}'' }
-      ''
-      + optionalString (!stdenv.isDarwin) ''
-        # copy icon and patch the original neovim.desktop file
-        mkdir -p $out/share/{applications,pixmaps}
-        ln -s ${neovim}/share/pixmaps/nvim.png $out/share/pixmaps/nvim.png
+      postBuild = ''
+        # Remove the symlinks created by symlinkJoin which we need to perform
+        # extra actions upon
+        rm $out/share/applications/nvim.desktop $out/bin/nvim
+        makeWrapper ${builtins.concatStringsSep " " makeWrapperArgs} ${extraMakeWrapperArgs}
         substitute ${neovim}/share/applications/nvim.desktop $out/share/applications/nvim.desktop \
           --replace 'TryExec=nvim' "TryExec=$out/bin/nvim" \
           --replace 'Name=Neovim' 'Name=WrappedNeovim'
       ''
       + optionalString withPython ''
         makeWrapper ${pythonEnv}/bin/python $out/bin/nvim-python --unset PYTHONPATH
-      '' + optionalString withPython3 ''
+      ''
+      + optionalString withPython3 ''
```

Comment:
is that a personal pref or enforced by a linter ?

Reply:
What do you mean? In anycase, all of my nix syntax habits are meant to optimize future diffs.